### ISSUE TITLE
disk plugin does not use udev

### ISSUE THREAD

**Issue by connection-reset** [collectd/collectd#2651]
*   Version of collectd: 5.8.0
*   Operating system / distribution: Arch Linux

## Expected behavior
Use udev attributes in disk plugin
```
LoadPlugin disk
<Plugin disk>
    UdevNameAttr "ID_SERIAL"
</Plugin>
```

## Actual behavior
disk plugin does not use udev
`disk plugin: The "UdevNameAttr" option is only supported if collectd is built with libudev support`

## Steps to reproduce

collectd is compiled with libsystemd 236.0 which provides libudev
from configure:
```
checking libudev.h usability... yes
checking libudev.h presence... yes
checking for libudev.h... yes
checking for udev_new in -ludev... yes
```

It works fine if I remove all the `HAVE_UDEV_H` checks from disk.c

**Comment by dstathis** [comment]
#2688

**Comment by rubenk** [comment]
Should be fixed by ed696c1b3adcb42d55759efe6f8034337f0fd243
